hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java

```diff
@@ -627,6 +638,7 @@ private boolean skipTokenRenewal(Token<?> token)
     if (identifier == null) {
       return false;
     }
+    expiredTime.set(identifier.getMaxDate());
     Text renewer = identifier.getRenewer();
     return (renewer != null && renewer.toString().equals(""));
```

Comment:
Just suggest to check if delegation token need to renew while app is just submitted.